### ISSUE TITLE
Build doesn't work when in a workspace and vendored

### ISSUE THREAD

**Issue by Awfa** [microsoft/windows-rs#691]
Trying to use this crate in a workspace where dependencies are vendored doesn't work.

Repro steps:
 - Make a cargo workspace with a project that depends on `windows` crate
 - Vendor dependencies into the workspace via `cargo vendor` + the required `config.toml` changes
 - Try building

The error seems to come from [this line](https://github.com/microsoft/windows-rs/blob/0e7b80a55cd76a2a6ab8af8bf0fac0e8eddc5c4f/crates/gen/src/workspace.rs#L39). 
It returns an empty string in this situation. Running the cargo metadata command manually reveals this error:
```
error: current package believes it's in a workspace when it's not:
current:   <>\rust_third_party\windows_gen-0.7.0\Cargo.toml
workspace: <>\Cargo.toml

this may be fixable by adding `rust_third_party\windows_gen-0.7.0` to the `workspace.members` array of the manifest located at: <>\Cargo.toml
Alternatively, to keep it out of the workspace, add the package to the `workspace.exclude` array, or add an empty `[workspace]` table to the package's manifest.
```
(rust_third_party is where I vendored the crates)

**Comment by rylev** [comment]
Hmmm... the error message seems to indicate that you haven't added `windows_gen-0.7.0` to the Config.toml's `workspace.members` array. The gen crate assumes it's part of the workspace.

**Comment by Awfa** [comment]
> Hmmm... the error message seems to indicate that you haven't added `windows_gen-0.7.0` to the Config.toml's `workspace.members` array. The gen crate assumes it's part of the workspace.

I believe that when you vendor dependencies with cargo vendor, dependencies should work just the same as when they aren't vendored (after the config.toml change). You shouldn't have to add the vendored dependency to the workspace members.

**Comment by kennykerr** [comment]
The [vendor](https://doc.rust-lang.org/cargo/commands/cargo-vendor.html) command vendors things... Recursive definitions are super helpful.😉

**Comment by Awfa** [comment]
Talked with @kennykerr over Teams.

This seems to be fixed in version 0.8!

Here's a repro repository: https://github.com/Awfa/windows-rs-issue-repro

`cargo build` when in the vtest_on_new_windows but not in the vtest_on_old_windows folder.
The 'new_windows' is using 0.8, and the 'old_windows' is using 0.7.

**Comment by rylev** [comment]
Awesome! Going to close this then. We can reopen it if the problem turns out to not be actually fixed.
